docs/src/main/mdoc/typeclasses/bimonad.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: docs
+title:  "Bimonad"
+section: "typeclasses"
+source: "core/src/main/scala/cats/Bimonad.scala"
+scaladoc: "#cats.Bimonad"
+---
+# Bimonad
+
+The `Bimonad` trait directly extends `Monad` and `Comonad` without introducing new methods. `Bimonad` is
+different from other `Bi` typeclasses like `Bifunctor`, `Bifoldable` or `Bitraverse` where the prefix describes a
+`F[_, _]`. The `Bimonad` is a `F[_]` and the `Bi` prefix has a different meaning here: it's both a `Monad` and a `Comonad`.
+
+
+If you use `Bimonad` as a convenience type such that:
+```scala
+def f[T[_] : Monad, Comonad, S](fa: T[S]): S
+```
+is re-written to:
+```scala
+def f[T[_] : Bimonad, S](fa: T[S]): S
+```
+keep in mind `Bimonad` has its own added laws so something that is both monadic
+and comonadic may not necessarily be a lawful `Bimonad`.
+
+###Eval as a Bimonad
+Eval is a lawful `Bimonad` so you can chain computations (like a `Monad`) and `extract` the result at the end (like a `Comonad`).
+
+Here is a possible implementation based on existing monad and comonad:
+```scala mdoc
+import cats._
+import cats.data._
+import cats.implicits._
+
+implicit def evalBimonad(implicit monad: Monad[Eval], comonad: Comonad[Eval]) =
+  new Bimonad[Eval] {
```

Comment:
`Eval[_]` is a bit of a bad choice here because the Comonad is not very safe. Calling `.value` on an Eval must be done with some care (it needs to be at O(1) stack depth or you lose the stack safety.  If you are calling this as part of a recursive call it often blows up).

`Id[_]` is also a Comonad, but a bit of a boring one. I wonder if `NonEmptyList[_]` would be a better example since it is non-trivial but also not somewhat unsafe in this way.

Reply:
Hm ok, I was hesitating between `Eval[_]` and `NonEmptyList[_]` for the main example.
I'll just use `NonEmptyList[_]`.